sources/api/bootstrap-containers/src/main.rs

```diff
@@ -0,0 +1,708 @@
+    let dependency = if essential { "RequiredBy" } else { "WantedBy" };
+
+    writeln!(output, "[Install]").context(error::WriteConfigurationValue { value: "[Install]" })?;
+    writeln!(output, "{}=configured.target", dependency)
+        .context(error::WriteConfigurationValue { value: dependency })?;
+    debug!("Writing drop-in file for {}", name);
+    fs::create_dir_all(&drop_in_dir).context(error::Mkdir { dir: &drop_in_dir })?;
+    fs::write(&drop_in_path, output)
+        .context(error::WriteConfigurationFile { path: drop_in_path })?;
+
+    Ok(())
+}
+
+/// Query the API for the currently defined bootstrap containers
+async fn get_bootstrap_containers<P>(
+    socket_path: P,
+) -> Result<HashMap<Identifier, model::BootstrapContainer>>
+where
+    P: AsRef<Path>,
+{
+    debug!("Querying the API for settings");
+
+    let method = "GET";
+    let uri = API_SETTINGS_URI;
+    let (_code, response_body) = apiclient::raw_request(&socket_path, uri, method, None)
+        .await
+        .context(error::APIRequest { method, uri })?;
+
+    // Build a Settings struct from the response string
+    debug!("Deserializing response");
+    let settings: model::Settings =
+        serde_json::from_str(&response_body).context(error::ResponseJson { method, uri })?;
+
+    // If bootstrap containers aren't defined, return an empty map
+    Ok(settings.bootstrap_containers.unwrap_or_default())
+}
+
+/// SystemdUnit stores the systemd unit being manipulated
+struct SystemdUnit<'a> {
+    unit: &'a str,
+}
+
+impl<'a> SystemdUnit<'a> {
+    fn new(unit: &'a str) -> Self {
+        SystemdUnit { unit }
+    }
+
+    fn is_enabled(&self) -> Result<bool> {
+        match command(SYSTEMCTL_BIN, &["is-enabled", &self.unit]) {
+            Ok(()) => Ok(true),
+            Err(e) => {
+                // If the systemd unit is not enabled, then `systemctl is-enabled` will return a
+                // non-zero exit code.
+                match e {
+                    error::Error::CommandFailure { .. } => Ok(false),
+                    _ => {
+                        // Otherwise, we return the error
+                        Err(e)
+                    }
+                }
+            }
+        }
+    }
+
+    fn is_active(&self) -> Result<bool> {
+        match command(SYSTEMCTL_BIN, &["is-active", &self.unit]) {
+            Ok(()) => Ok(true),
+            Err(e) => {
+                // If the systemd unit is not active(running), then `systemctl is-active` will
+                // return a non-zero exit code.
+                match e {
+                    error::Error::CommandFailure { .. } => Ok(false),
+                    _ => {
+                        // Otherwise, we return the error
+                        Err(e)
+                    }
+                }
+            }
+        }
+    }
+
+    fn enable(&self) -> Result<()> {
+        // Only enable the unit, since it will be started once systemd reaches the `configured`
+        // target
```

Comment:
nit: the unit is started after systemd reaches the `preconfigured` target

Reply:
Fixed